### DISCUSSION TITLE
How to display images larger than frame?

### DISCUSSION THREAD

**Discussion by Lydanne** [fltk-rs/fltk-rs#1250] [Q&A]
How to display images larger than frame?

After I take a screenshot, I want to display the image, but now only a part of it can be displayed, but the img.scale  the image can fully display, but the displayed image is clear, so I don't want to use this.

Is there a way to display the original image?

![image](https://user-images.githubusercontent.com/39021696/179497792-8fed2015-6ddd-4182-a278-857470e175ed.png)

repo: 
https://github.com/WumaCoder/fltk-demo

**Comment by MoAlyousef** [comment]
To display the whole image you either need to rescale the image using img.scale(). You can try passing `true` for the 3rd argument (proportional). Or you can resize the frame to be the same size as the image, before setting the image.

**Reply by Lydanne** [reply]
I now this picture is much larger than the frame, if I adjust the scale, it will be unclear.

**Reply by Lydanne** [reply]
I was wondering if it could be displayed in full resolution

**Reply by MoAlyousef** [reply]
You can try changing the scaling algorithm:
```rust
fltk::image::RgbImage::set_scaling_algorithm(fltk::image::RgbScaling::Bilinear);
```

**Reply by Lydanne** [reply]
It's still unclear.

code:
![image](https://user-images.githubusercontent.com/39021696/179650593-e38370a5-b05e-4ecd-9183-1e204448bf1d.png)


raw image:
![image](https://user-images.githubusercontent.com/39021696/179650466-3e7f8447-1d4f-4444-90e7-a3e56eda673f.png)

scale image:
![image](https://user-images.githubusercontent.com/39021696/179650523-78383845-c52b-40c0-90e8-3807645d4ddc.png)

**Reply by MoAlyousef** [reply]
You can try to resize first thru the image crate itself to see if that makes a difference. When you resize an image to a smaller image then again to a larger image, there will be loss of information.

**Reply by Lydanne** [reply]
let me try
